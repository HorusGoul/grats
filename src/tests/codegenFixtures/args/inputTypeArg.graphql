input Greeting {
  salutation: String!
  name: String!
}

type Query {
  greet(greeting: Greeting): String
}
