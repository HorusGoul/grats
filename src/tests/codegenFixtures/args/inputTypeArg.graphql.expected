-----------------
INPUT
----------------- 
input Greeting {
  salutation: String!
  name: String!
}

type Query {
  greet(greeting: Greeting): String
}

-----------------
OUTPUT
-----------------
import { GraphQLSchema, GraphQLObjectType, GraphQLString, GraphQLInputObjectType, GraphQLNonNull } from "graphql";
const GreetingType: GraphQLInputObjectType = new GraphQLInputObjectType({
    name: "Greeting",
    fields() {
        return {
            salutation: {
                name: "salutation",
                type: new GraphQLNonNull(GraphQLString)
            },
            name: {
                name: "name",
                type: new GraphQLNonNull(GraphQLString)
            }
        };
    }
});
const QueryType: GraphQLObjectType = new GraphQLObjectType({
    name: "Query",
    fields() {
        return {
            greet: {
                name: "greet",
                type: GraphQLString,
                args: {
                    greeting: {
                        name: "greeting",
                        type: GreetingType
                    }
                }
            }
        };
    }
});
const schema = new GraphQLSchema({
    query: QueryType
});
export { schema };
